R/unnest-longer.R

```diff
@@ -261,10 +259,20 @@ collect_indices_info <- function(col, indices_include) {
   } else {
     # Indices are requested, and some elements are named.
     # For any unnamed elements, generate `NA` indices.
-    indices[unnamed] <- map(sizes[unnamed], vec_rep, x = NA_character_)
+    indices[unnamed] <- map(sizes[unnamed], vec_rep, x = "")
     index_ptype <- character()
   }
 
+  if (keep_empty) {
+    empty <- sizes == 0L
+
+    if (any(empty)) {
+      # `NULL` or typed empty elements get an `NA` index of the right type
+      index_empty <- vec_init(index_ptype)
+      indices[empty] <- list(index_empty)
```

Comment:
And here is the change to consistently use `NA` of the right type for `NULL` or typed empty elements that we "keep" (rather than using a mix of `1L` and `NA_character_` somewhat inconsistently)